packages/flutter/lib/src/material/switch.dart

```diff
@@ -421,11 +421,12 @@ class Switch extends StatelessWidget {
         height: size.height,
         alignment: Alignment.center,
         child: CupertinoSwitch(
-            dragStartBehavior: dragStartBehavior,
-            value: value,
-            onChanged: onChanged,
-            activeColor: activeColor,
-            trackColor: inactiveTrackColor,
+          dragStartBehavior: dragStartBehavior,
+          value: value,
+          onChanged: onChanged,
+          activeColor: activeColor,
+          trackColor: inactiveTrackColor,
+          thumbColor: inactiveThumbColor,
```

Comment:
Is this the correct color to map to? It's this supposed to be `thumbColor`?

Reply:
I assumed it was like `trackColor` and `inactiveTrackColor` just above. But if not, shouldn't we map `trackColor` with `trackColor` too?

Reply:
I think you're right.

```dart
  /// The color to use on the thumb when this switch is off.
  ///
  /// Defaults to the colors described in the Material design specification.
  ///
  /// Ignored if this switch is created with [Switch.adaptive].
  ///
  /// If [thumbColor] returns a non-null color in the default state, it will be
  /// used instead of this color.
  final Color? inactiveThumbColor;

  /// The color to use on the track when this switch is off.
  ///
  /// Defaults to the colors described in the Material design specification.
  ///
  /// Ignored if this switch is created with [Switch.adaptive].
  ///
  /// If [trackColor] returns a non-null color in the default state, it will be
  /// used instead of this color.
  final Color? inactiveTrackColor;
```

The doc says inactive color properties are ignored if switch is created with `Switch.adaptive`. So I think we should also change the line above.

Reply:
The problem is that `thumbColor` and `trackColor` are MaterialStateProperty and we need a Color. If we keep on with this solution, in what `MaterialState` should we resolve them?

Reply:
Nevermind, inactive colors are supposed to be used as is. This saves us from having to draw a Color from a MaterialStateProperty.

```dart
/// If null, then the value of [activeColor] is used in the selected
/// state and [inactiveThumbColor] in the default state. If that is also null,
/// then the value of [SwitchThemeData.thumbColor] is used. If that is also
/// null, then the following colors are used:
///
/// | State    | Light theme                       | Dark theme                        |
/// |----------|-----------------------------------|-----------------------------------|
/// | Default  | `Colors.grey.shade50`             | `Colors.grey.shade400`            |
/// | Selected | [ThemeData.toggleableActiveColor] | [ThemeData.toggleableActiveColor] |
/// | Disabled | `Colors.grey.shade400`            | `Colors.grey.shade800`            |
final MaterialStateProperty<Color?>? thumbColor;
```